### ISSUE TITLE
wit-bindgen go: support resources

### ISSUE THREAD

**Issue by pchickey** [bytecodealliance/wit-bindgen#696]

**Comment by Mossaka** [comment]
Duplicate of #615

**Comment by Mossaka** [comment]
I've started working on adding runtime tests for rust and C to the repo. Once this is done I will start implementing resources in Go and write more tests!
